src/session/content/message_row/text.rs

```diff
@@ -200,6 +205,37 @@ impl MessageText {
             unreachable!("Unexpected message type: {:?}", message);
         }
 
+        if let Some(message) = message.downcast_ref::<Message>() {
+            if message.reply_to_message_id() != 0 {
+                if let Some(replied_message) = message
+                    .chat()
+                    .history()
+                    .message_by_id(message.reply_to_message_id())
+                {
+                    imp.replied_message.replace(Some(replied_message));
+                } else {
+                    spawn(clone!(@weak self as obj, @weak message => async move {
+                        let chat = message.chat();
+                        let result = functions::get_message(chat.id(), message.id(), chat.session().client_id()).await;
+                        if let Ok(e_message) = result {
+                            let EnumMessage::Message(message) = e_message;
+                            let message = Message::new(message, &chat);
+                            obj.imp().replied_message.replace(Some(message));
+                    }}));
+                }
+                if let Some(replied_message) = imp.replied_message.borrow().clone() {
+                    imp.reply_message
+                        .set_message(&replied_message, message.is_outgoing());
+                }
```

Comment:
This is not correct. The `spawn` function will spawn an async future, so you can't just put an if below and expect that the value has been set by the async call in-place. What you should do here is something like:
```suggestion
                    imp.reply_message.set_message(&replied_message, message.is_outgoing());
                } else {
                    spawn(clone!(@weak self as obj, @weak message => async move {
                        let chat = message.chat();
                        let result = functions::get_message(chat.id(), message.id(), chat.session().client_id()).await;
                        if let Ok(e_message) = result {
                            let EnumMessage::Message(message) = e_message;
                            let replied_message = Message::new(message, &chat);
                            obj.imp().reply_message.set_message(&replied_message, message.is_outgoing());
                    }}));
                }
```

Reply:
Oh, and the `get_message` needs `message.reply_to_message_id()` instead of `message.id()`.